tests/integration-tests/clusters_factory.py

```diff
@@ -30,11 +30,35 @@ def __init__(self, name, config_file, ssh_key):
         self.__cfn_outputs = None
         self.__cfn_resources = None
 
-    def update(self):
-        """Rewrite the configuration file starting from the self.config object."""
+    def update(self, reset_desired=None, extra_params=None):
+        """
+        Update a cluster with an already updated config.
+        :param reset_desired: reset the current ASG desired capacity to initial config values
+        :param extra_params: extra parameters to pass to stack update
+        """
+        # Rewrite the configuration file starting from the already updated self.config object
```

Comment:
since the self.config is updated in a separate method shall we rewrite the file where the config is updated? For instance in _update_cluster_property?

Reply:
Moved in the `_update_cluster` to avoid to repeat the write multiple times in the `_update_cluster_property`